event/consume.go

```diff
@@ -44,6 +44,7 @@ func (p *Processor) Consume(ctx context.Context, cg kafka.IConsumerGroup, h Hand
 				}
 
 				msg.Release()
+				cancel()
```

Comment:
Should this be done in a `defer` call to ensure it is always called?
```
msgCtx, cancel := context.WithCancel(ctx)
defer func() {
    msg.Release()
    cancel()
}()
```
I think it is OK as is assuming that `processMessage` doesn't panic and this code doesn't change in future. I think if there was a panic it wouldn't cancel or release the message. 

Using a defer call offers that additional safety to guarantee any clean up is always executed. Possibly I am paranoid but I think it's a good approach to adopt plus it's also in the [Go context Cancel example doc](https://pkg.go.dev/context#WithCancel)

Reply:
Thanks for review.
If the code panics, the app will bomb out of the for {} ever kafka message consume loop and will terminate, so strictly speaking a `defer cancel()`  would not help in this instance, but there is a bigger problem with what you suggest ...

defer's get called when a function exits, so your suggestion will stop the code working as the msg.Release() never gets called, and thus the cancel() would also not get called.

I am happy that my fix to context memory leaks is technically correct.

Reply:
Good point - I didn't realise it was inside a loop so as you say defer would never actually be called. In that case approved.